lib/rubocop/cop/naming/inclusive_language.rb

```diff
@@ -182,12 +184,16 @@ def process_allowed_regex(allowed)
           Array(allowed).map do |allowed_term|
             next if allowed_term.is_a?(String) && allowed_term.strip.empty?
 
-            ensure_regex_string(allowed_term)
+            ensure_case_insensitive_regexp(allowed_term)
           end
         end
 
-        def ensure_regex_string(regex)
-          regex.is_a?(Regexp) ? regex.source : regex
+        def ensure_case_insensitive_regexp(object)
+          case object
+          when Regexp then Regexp.new(object.source, object.options | Regexp::IGNORECASE)
```

Comment:
Could optimize to skip creating duplicates

```suggestion
          when Regexp
            return object if object.options == Regexp::IGNORECASE

            Regexp.new(object.source, object.options | Regexp::IGNORECASE)
```